pkg/test/utils/kubernetes.go

```diff
@@ -395,13 +394,14 @@ func ConvertUnstructured(in runtime.Object) (runtime.Object, error) {
 	if group == kudoGroup {
 		log.Print("WARNING: kudo.dev group has been deprecated and is schedule to be removed in KUTTL 0.8.0")
 	}
-	if (group == kudoGroup || group == kuttlGroup) && kind == "TestStep" {
+	switch {
+	case (group == kudoGroup || group == kuttlGroup) && kind == "TestStep":
 		converted = &harness.TestStep{}
-	} else if (group == kudoGroup || group == kuttlGroup) && kind == "TestAssert" {
+	case (group == kudoGroup || group == kuttlGroup) && kind == "TestAssert":
 		converted = &harness.TestAssert{}
-	} else if (group == kudoGroup || group == kuttlGroup) && kind == "TestSuite" {
+	case (group == kudoGroup || group == kuttlGroup) && kind == "TestSuite":
 		converted = &harness.TestSuite{}
-	} else {
+	default:
 		return in, nil
 	}
```

Comment:
I wouldn't mind extracting the `(group == kudoGroup || group == kuttlGroup)` part into a separate "if" before the switch, so end up with a simpler switch:

```
if !(group == kudoGroup || group == kuttlGroup) {
  return in, nil
}
switch kind {
  case "TestStep:
    converted = &harness.TestStep{}
  case "TestAssert:
   converted = &harness.TestAssert{}
  case "TestSuite":
    converted = &harness.TestSuite{}
  default:
    return in, nil
}
```

Reply:
that makes a lot more sense!  thx